crates/world/payload/src/builder.rs

```diff
@@ -624,7 +624,20 @@ where
                     best_txs.mark_invalid(tx.signer(), tx.nonce());
                     continue;
                 }
+                // Step 1: Collect all nullifier_hashes for this transaction
+                let mut same_tx_nullifiers = HashSet::new();
+                let has_duplicate_nullifier_in_same_tx = payloads
+                    .iter()
+                    .any(|payload| !same_tx_nullifiers.insert(payload.nullifier_hash));
+
+                // Step 2: If there are duplicates within the transaction, mark as invalid and don't add to spent_nullifier_hashes
+                if has_duplicate_nullifier_in_same_tx {
+                    best_txs.mark_invalid(tx.signer(), tx.nonce());
+                    invalid_txs.push(*pooled_tx.hash());
+                    continue;
+                }
 
+                // Step 3: Check for duplicates across all transactions.
```

Comment:
Wdyt about updating the PBH validation logic to reject duplicate nullifier hashes in the same bundle? That way, we can avoid making any changes to the payload building logic to address the issue.